AlphaWallet/Tokens/Logic/SuggestEIP1559.swift

```diff
@@ -0,0 +1,209 @@
+//
+//  SuggestEIP1559.swift
+//  AlphaWallet
+//
+//  Created by Vladyslav Shepitko on 16.08.2022.
+//
+
+import Foundation
+import BigInt
+import PromiseKit
+import web3swift
+
+struct SuggestEIP1559 {
+    /// priority fee offered when there are no recent transactions
+    private let fallbackPriorityFee: BigInt = .init(2000000000)
+
+    /// effective reward value to be selected from each individual block
+    private let rewardPercentile = [10]
+
+    /// suggested priority fee to be selected from sorted individual block reward percentiles
+    private let rewardBlockPercentile = 40
+
+    /// highest timeFactor index in the returned list of suggestion
+    private let maxTimeFactor: Int = 15
+
+    /// sampled percentile range of exponentially weighted baseFee history
+    private let sampleMinPercentile: Double = 10
+
+    private let sampleMaxPercentile: Double = 30
+
+    /// extra priority fee offered in case of expected baseFee rise
+    private let extraPriorityFeeRatio: NSDecimalNumber = NSDecimalNumber(value: 0.25)
+
+    let feeHistory: FeeHistory
+
+    func suggestPriorityFee(firstBlock: BigInt, gasUsedRatio: [Double], gasService: GetFeeHistory) throws -> BigInt {
+        var ptr = gasUsedRatio.count - 1
+        var needBlocks = 5
+        var rewards: [BigInt] = []
+        while needBlocks > 0 && ptr >= 0 {
+            let blockCount = maxBlockCount(gasUsedRatio: gasUsedRatio, ptr, needBlocks)
+            if blockCount > 0 {
+                // feeHistory API call with reward percentile specified is expensive and therefore is only requested for a few non-full recent blocks.
+                let lastBlock = firstBlock + BigInt(ptr)
+
+                print("[GasEstimator] lastBlock: \(lastBlock.hexEncoded) for ptr: \(ptr), c: \(BigInt(lastBlock.hexEncoded.drop0x, radix: 16))")
+
+                let feeHistory = try gasService.feeHistory(blockCount, lastBlock: lastBlock.hexEncoded, rewardPercentile: rewardPercentile).wait()
+                let rewardSize = feeHistory.reward?.count ?? 0
+                if rewardSize > 0 {
```

Comment:
Think this `> 0` isn't necessary?

Reply:
moved android implementation of  `SuggestEIP1559` into swift

Reply:
the is for loop right after this check `for index in 0 ..< rewardSize {`  in case if there is 0 in rewardSize, it will cause a crash.

Reply:
If `rewardSize` is 0, it will skip the for-loop, right?

Reply:
Yep